src/scenario/condition_types/condition_types.c

```diff
@@ -362,6 +374,34 @@ int scenario_condition_type_time_met(const scenario_condition_t *condition)
     return comparison_helper_compare_values(comparison, total_months, target_months);
 }
 
+int scenario_condition_type_trade_route_open_met(const scenario_condition_t *condition)
+{
+    int route_id = condition->parameter1;
+    int check_for_open = condition->parameter2;
+
+    if (!trade_route_is_valid(route_id)) {
+        return 0;
+    }
+
+    int route_is_open = empire_city_is_trade_route_open(route_id);
+    return (route_is_open == check_for_open);
+}
+
+int scenario_condition_type_trade_route_price_met(const scenario_condition_t *condition)
+{
+    int route_id = condition->parameter1;
+    int comparison = condition->parameter2;
+    int value = condition->parameter3;
+
+    if (!trade_route_is_valid(route_id)) {
+        return 0;
+    }
+
+    int route_is_open = empire_city_is_trade_route_open(route_id);
+    int route_price = empire_city_get_trade_route_cost(route_id);
+    return route_is_open ? 0 : route_price;
```

Comment:
Aren't you missing the actual check for the condition here?

Reply:
Fixed.